code/src/reducers/todos.js

```diff
@@ -0,0 +1,35 @@
+import { createSlice } from "@reduxjs/toolkit";
+import uniqid from 'uniqid'
+
+const initialState = {
+  items: []
+}
+
+const todos = createSlice({
+  name: 'todos',
+  initialState,
+  reducers: {
+    addToDo: (state, action) => {
+      state.items.unshift({
+        id: uniqid(),
+        task: action.payload.input,
+        isComplete: false,
+        newDate: new Date(),
+        category: action.payload.category
+      })
+    },
+    removeToDo: (state, action) => {
+      state.items = state.items.filter((item) => item.id !== action.payload)
+    },
+    toggleIsComplete: (state, action) => {
+      const task = state.items.find((item) => item.id === action.payload)
+      task.isComplete = !task.isComplete
+      state.items = state.items.sort((a,b) => a.isComplete - b.isComplete)
```

Comment:
Can you maybe leave comments about your functions and what they do? I see that you filtering items array, and then you are setting task from true to false on line 26, is that correct? Then you doing sorting on items array with task that is false, or ?

Reply:
Sorry, you are right, I should be better with the commenting!
Line 26 has nothing to do with Line 27, except that they are both happening, when the user clicks the checkbox. 
So line 25/26 just take care of the store getting updated every time, that the user clicks the checkbox of a task.
Line 27 sorts the array of items by the key "isComplete", so that the completed items move to the end